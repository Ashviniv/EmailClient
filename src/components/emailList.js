import React from "react";
import { Input, Button, Table, Card, CardBody, CardTitle, CardSubtitle } from "reactstrap";

export const EmailList = (props) => {
  return (
    <Table>
      <tbody>
      {
        props.emails.map(email => {
          return <EmailRow {...email} key={email.id} onClickRow={props.onClickRow} />;
        })
      }
      </tbody>
    </Table>
  )
}

const EmailRow = (props) => {
  return (
    <tr
      key={props.id}
      onClick={props.onClickRow(props)}>
      <td>
        <Input type="checkbox" />
      </td>
      <td>
        <h4><a href="/">{ props.title }</a></h4>
      </td>
      <td>
        <Button color="primary">R</Button>
      </td>
    </tr>
  )
}

export const EmailPreview = (props) => {
  return (
    <Card>
      <CardBody>
        <CardTitle>
          { props.email.title }
        </CardTitle>
        <CardSubtitle>
          { props.email.body }
        </CardSubtitle>
      </CardBody>
    </Card>
  )
}