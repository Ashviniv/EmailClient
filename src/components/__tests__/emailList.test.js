import React from "react";
import { shallow } from "../../enzyme.js";
import { EmailList } from "../emailList.js";

const emailList = [
  {
    title: "monit alert -- Exists delayed_job",
    id: 1,
    body: <h6>monit alert -- Exists delayed_job</h6>
  },
  {
    title: "monit alert -- Exists delayed_job 2",
    id: 2,
    body: <h6>monit alert -- Exists delayed_job 2</h6>
  },
  {
    title: "monit alert -- Exists delayed_job 3",
    id: 3,
    body: "<h1>monit alert -- Exists delayed_job 3</h1>"
  },
  {
    title: "monit alert -- Exists delayed_job 4",
    id: 4,
    body: "<h1>monit alert -- Exists delayed_job 4</h1>"
  }
];

const onClickRow = jest.fn()

describe("<EmailList />", () => {
  it("Must render emailList component", () => {

    const component = shallow(<EmailList
      emails={emailList}
      onClickRow={onClickRow}
    />)

    expect(component).toMatchSnapshot()
  })
})