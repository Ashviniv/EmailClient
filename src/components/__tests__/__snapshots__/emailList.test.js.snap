// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailList /> Must render emailList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    emails={
        Array [
            Object {
              "body": <h6>
                monit alert -- Exists delayed_job
        </h6>,
              "id": 1,
              "title": "monit alert -- Exists delayed_job",
            },
            Object {
              "body": <h6>
                monit alert -- Exists delayed_job 2
        </h6>,
              "id": 2,
              "title": "monit alert -- Exists delayed_job 2",
            },
            Object {
              "body": "<h1>monit alert -- Exists delayed_job 3</h1>",
              "id": 3,
              "title": "monit alert -- Exists delayed_job 3",
            },
            Object {
              "body": "<h1>monit alert -- Exists delayed_job 4</h1>",
              "id": 4,
              "title": "monit alert -- Exists delayed_job 4",
            },
          ]
    }
    onClickRow={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <tbody>
        <EmailRow
                body={
                        <h6>
                                monit alert -- Exists delayed_job
                        </h6>
                }
                id={1}
                onClickRow={[Function]}
                title="monit alert -- Exists delayed_job"
        />
        <EmailRow
                body={
                        <h6>
                                monit alert -- Exists delayed_job 2
                        </h6>
                }
                id={2}
                onClickRow={[Function]}
                title="monit alert -- Exists delayed_job 2"
        />
        <EmailRow
                body="<h1>monit alert -- Exists delayed_job 3</h1>"
                id={3}
                onClickRow={[Function]}
                title="monit alert -- Exists delayed_job 3"
        />
        <EmailRow
                body="<h1>monit alert -- Exists delayed_job 4</h1>"
                id={4}
                onClickRow={[Function]}
                title="monit alert -- Exists delayed_job 4"
        />
</tbody>,
      "responsiveTag": "div",
      "tag": "table",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <EmailRow
            body={
                        <h6>
                                    monit alert -- Exists delayed_job
                        </h6>
            }
            id={1}
            onClickRow={[Function]}
            title="monit alert -- Exists delayed_job"
/>,
          <EmailRow
            body={
                        <h6>
                                    monit alert -- Exists delayed_job 2
                        </h6>
            }
            id={2}
            onClickRow={[Function]}
            title="monit alert -- Exists delayed_job 2"
/>,
          <EmailRow
            body="<h1>monit alert -- Exists delayed_job 3</h1>"
            id={3}
            onClickRow={[Function]}
            title="monit alert -- Exists delayed_job 3"
/>,
          <EmailRow
            body="<h1>monit alert -- Exists delayed_job 4</h1>"
            id={4}
            onClickRow={[Function]}
            title="monit alert -- Exists delayed_job 4"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "body": <h6>
              monit alert -- Exists delayed_job
</h6>,
            "id": 1,
            "onClickRow": [Function],
            "title": "monit alert -- Exists delayed_job",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "body": <h6>
              monit alert -- Exists delayed_job 2
</h6>,
            "id": 2,
            "onClickRow": [Function],
            "title": "monit alert -- Exists delayed_job 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "body": "<h1>monit alert -- Exists delayed_job 3</h1>",
            "id": 3,
            "onClickRow": [Function],
            "title": "monit alert -- Exists delayed_job 3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "body": "<h1>monit alert -- Exists delayed_job 4</h1>",
            "id": 4,
            "onClickRow": [Function],
            "title": "monit alert -- Exists delayed_job 4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "tbody",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <tbody>
          <EmailRow
                    body={
                              <h6>
                                        monit alert -- Exists delayed_job
                              </h6>
                    }
                    id={1}
                    onClickRow={[Function]}
                    title="monit alert -- Exists delayed_job"
          />
          <EmailRow
                    body={
                              <h6>
                                        monit alert -- Exists delayed_job 2
                              </h6>
                    }
                    id={2}
                    onClickRow={[Function]}
                    title="monit alert -- Exists delayed_job 2"
          />
          <EmailRow
                    body="<h1>monit alert -- Exists delayed_job 3</h1>"
                    id={3}
                    onClickRow={[Function]}
                    title="monit alert -- Exists delayed_job 3"
          />
          <EmailRow
                    body="<h1>monit alert -- Exists delayed_job 4</h1>"
                    id={4}
                    onClickRow={[Function]}
                    title="monit alert -- Exists delayed_job 4"
          />
</tbody>,
        "responsiveTag": "div",
        "tag": "table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <EmailRow
              body={
                            <h6>
                                          monit alert -- Exists delayed_job
                            </h6>
              }
              id={1}
              onClickRow={[Function]}
              title="monit alert -- Exists delayed_job"
/>,
            <EmailRow
              body={
                            <h6>
                                          monit alert -- Exists delayed_job 2
                            </h6>
              }
              id={2}
              onClickRow={[Function]}
              title="monit alert -- Exists delayed_job 2"
/>,
            <EmailRow
              body="<h1>monit alert -- Exists delayed_job 3</h1>"
              id={3}
              onClickRow={[Function]}
              title="monit alert -- Exists delayed_job 3"
/>,
            <EmailRow
              body="<h1>monit alert -- Exists delayed_job 4</h1>"
              id={4}
              onClickRow={[Function]}
              title="monit alert -- Exists delayed_job 4"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "body": <h6>
                monit alert -- Exists delayed_job
</h6>,
              "id": 1,
              "onClickRow": [Function],
              "title": "monit alert -- Exists delayed_job",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "body": <h6>
                monit alert -- Exists delayed_job 2
</h6>,
              "id": 2,
              "onClickRow": [Function],
              "title": "monit alert -- Exists delayed_job 2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "function",
            "props": Object {
              "body": "<h1>monit alert -- Exists delayed_job 3</h1>",
              "id": 3,
              "onClickRow": [Function],
              "title": "monit alert -- Exists delayed_job 3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "function",
            "props": Object {
              "body": "<h1>monit alert -- Exists delayed_job 4</h1>",
              "id": 4,
              "onClickRow": [Function],
              "title": "monit alert -- Exists delayed_job 4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
